Samples/OIDCMigration/OIDCMigration/OIDCSignInViewController.swift

```diff
@@ -0,0 +1,136 @@
+//
+// Copyright (c) 2022-Present, Okta, Inc. and/or its affiliates. All rights reserved.
+// The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+//
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and limitations under the License.
+//
+
+import UIKit
+import OktaOidc
+import AuthFoundation
+
+class OIDCSignInViewController: UIViewController {
+    @IBOutlet weak var signInButton: UIButton!
+    @IBOutlet weak var signOutButton: UIButton!
+    @IBOutlet weak var clientIdLabel: UILabel!
+    @IBOutlet weak var migrateButton: UIButton!
+    @IBOutlet weak var openUserProfileButton: UIButton!
+    
+    var oktaOidc: OktaOidc?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        updateUIState()
+        
+        // Dismiss the profile view controller when it is signed out
+        NotificationCenter.default.addObserver(forName: .defaultCredentialChanged, object: nil, queue: .main) { note in
+            guard note.object == nil,
+                  self.presentedViewController != nil
+            else { return }
+            
+            self.presentedViewController?.dismiss(animated: true, completion: {
+                self.updateUIState()
+            })
+        }
+    }
+    
+    func updateUIState() {
+        if let client = try? OktaOidc() {
+            oktaOidc = client
+            clientIdLabel.text = client.configuration.clientId
+            
+            if OktaOidcStateManager.readFromSecureStorage(for: client.configuration) != nil {
+                signInButton.isHidden = true
+                signOutButton.isHidden = false
+            } else {
+                signInButton.isHidden = false
+                signOutButton.isHidden = true
+            }
+        } else {
+            clientIdLabel.text = "Not configured"
+            signInButton.isEnabled = false
+            signOutButton.isHidden = true
```

Comment:
can we extract these into function, in this way we can reuse

Reply:
There doesn't seem to be much of a need for reuse within the sample apps. My intention was to keep the flow clear to understand, and to avoid adding additional complexity that could distract from the sample itself.